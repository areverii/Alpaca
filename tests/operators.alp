function void main(<string> args):
    int x

    prints("1 + 1")
    printi(1 + 1)

    prints("\n3 - 2")
    printi(3 - 2)

    prints("\n4 * 4")
    printi(4 * 4)

    prints("\n10 / 3")
    printi(10 / 3)

    prints("\n10 % 3")
    printi(10 % 3)

    prints("\n10 / -2")
    printi(10 / -2)

    prints("16 >> 1")
    printi(16 >> 1)

    prints("-16 >> 1")
    printi(-16 >> 1)

    prints("4 << 2")
    printi(4 << 2)

    prints("5 & 3")
    printi(5 & 3)

    prints("5 | 3")
    printi(5 | 3)

    prints("5 ^ 3")
    printi(5 ^ 3)

    x = 3
    prints("-x")
    printi(-x)



    prints("3.2 + 1")
    printf(3.2 + 1)

    prints("1 + 3.2")
    printf(1 + 3.2)

    prints("3.2 + 3.2")
    printf(3.2 + 3.2)


    
    prints("3.2 * 1")
    printf(3.2 * 1)

    prints("1 * 3.2")
    printf(1 * 3.2)

    prints("3.2 * 3.2")
    printf(3.2 * 3.2)


    
    prints("3.2 / 1")
    printf(3.2 / 1)

    prints("10 / 2.5")
    printf(10 / 2.5)

    prints("3.2 / 3.2")
    printf(3.2 / 3.2)


    
    prints("true and true")
    printb(true and true)

    prints("true and false")
    printb(true and false)

    prints("true or false")
    printb(true or false)

    prints("not false")
    printb(not false)


    
    prints("1.0 == 1.0")
    printb(1.0 == 1.0)

    prints("1.0 != 1.0")
    printb(1.0 != 1.0)

    prints("1 == 1.0")
    printb(1 == 1.0)

    prints("1 != 1.0")
    printb(1 != 1.0)


    
    prints("true == true")
    printb(true == true)

    prints("true != true")
    printb(true != true)

    prints("\"test\" == \"test\"")
    printb("test" == "test")

    prints("\"test\" == \"test1\"")
    printb("test" == "test1")

    prints("\"test\" != \"test\"")
    printb("test" != "test")

    prints("\"test\" != \"test1\"")
    printb("test" != "test1")

    prints("1 == 1")
    printb(1 == 1)

    prints("1 != 1")
    printb(1 != 1)

    prints("1 == 2")
    printb(1 == 2)

    prints("1 != 2")
    printb(1 != 2)

    prints("1 < 2")
    printb(1 < 2)

    prints("1 < 1")
    printb(1 < 1)

    prints("1 <= 2")
    printb(1 <= 2)

    prints("1 >= 2")
    printb(1 >= 2)

    prints("2 > 1")
    printb(2 > 1)

    prints("1 > 2")
    printb(1 > 2)

    prints("2.1 > 2")
    printb(2.1 > 2)

    prints("2.1 >= 2")
    printb(2.1 >= 2)

    prints("2.1 < 2")
    printb(2.1 < 2)

    prints("2.1 <= 2")
    printb(2.1 <= 2)

    prints("2.1 <= 2.1")
    printb(2.1 <= 2.1)

    prints("2.1 >= 2.1")
    printb(2.1 >= 2.1)

    prints("2.1 == 2.1")
    printb(2.1 == 2.1)

    prints("2.0 == 2")
    printb(2.0 == 2)

end
