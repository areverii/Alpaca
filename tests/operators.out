1 + 1
2

3 - 2
1

4 * 4
16

10 / 3
3

10 % 3
1

10 / -2
-5
16 >> 1
8
-16 >> 1
-8
4 << 2
16
5 & 3
1
5 | 3
7
5 ^ 3
6
-x
-3
3.2 + 1
4.2
1 + 3.2
4.2
3.2 + 3.2
6.4
3.2 * 1
3.2
1 * 3.2
3.2
3.2 * 3.2
10.24
3.2 / 1
3.2
10 / 2.5
4
3.2 / 3.2
1
true and true
1
true and false
0
true or false
1
not false
1
1.0 == 1.0
1
1.0 != 1.0
0
1 == 1.0
1
1 != 1.0
0
true == true
1
true != true
0
"test" == "test"
1
"test" == "test1"
0
"test" != "test"
0
"test" != "test1"
1
1 == 1
1
1 != 1
0
1 == 2
0
1 != 2
1
1 < 2
1
1 < 1
0
1 <= 2
1
1 >= 2
0
2 > 1
1
1 > 2
0
2.1 > 2
1
2.1 >= 2
1
2.1 < 2
0
2.1 <= 2
0
2.1 <= 2.1
1
2.1 >= 2.1
1
2.1 == 2.1
1
2.0 == 2
1
